modules/core/src/main/java/org/apache/fluo/core/oracle/OracleServer.java

```diff
@@ -195,18 +195,25 @@ public OracleServer(Environment env) throws Exception {
 
   private void allocateTimestamp() throws Exception {
     Stat stat = new Stat();
-    byte[] d = curatorFramework.getData().storingStatIn(stat).forPath(maxTsPath);
+    // FLUO-1000
+    synchronized (this) {
```

Comment:
Please correct me if I am wrong but I was under the impression that object locks do not inherit from my reading of Doug Lea's Concurrent Programming.
https://stackoverflow.com/questions/9525882/if-a-synchronized-method-calls-another-non-synchronized-method-is-there-a-lock

Reply:
That post is correct.  However in this situation, the method is private and only called by a synchronized public method.  Therefore nothing external can directly call the method in an unsynchronized way.    It would be fine to make the whole method synchronized if you think that makes the code more clear.